packages/react-components/react-tree/src/components/TreeItem/useTreeItem.ts

```diff
@@ -133,103 +129,70 @@ export function useTreeItem_unstable<Value = string>(
       case ArrowLeft:
         return handleArrowLeft(event);
       case End:
-        return requestNavigation({ event, value, type: treeDataTypes.end, target: event.currentTarget });
       case Home:
-        return requestNavigation({ event, value, type: treeDataTypes.home, target: event.currentTarget });
       case ArrowUp:
-        return requestNavigation({ event, value, type: treeDataTypes.arrowUp, target: event.currentTarget });
       case ArrowDown:
-        return requestNavigation({ event, value, type: treeDataTypes.arrowDown, target: event.currentTarget });
+        return requestNavigation({ event, value, type: treeDataTypes[event.key], target: event.currentTarget });
     }
     const isTypeAheadCharacter =
       event.key.length === 1 && event.key.match(/\w/) && !event.altKey && !event.ctrlKey && !event.metaKey;
     if (isTypeAheadCharacter) {
-      return requestNavigation({ event, value, target: event.currentTarget, type: treeDataTypes.typeAhead });
+      return requestNavigation({ event, value, target: event.currentTarget, type: treeDataTypes.TypeAhead });
     }
   });
 
-  const [isActionsVisible, setActionsVisible] = React.useState(false);
-  const showActions = useEventCallback((event: React.SyntheticEvent) => {
-    const isEventFromSubtree = subtreeRef.current && elementContains(subtreeRef.current, event.target as Node);
-    if (!isEventFromSubtree) {
+  const handleActionsVisible = useEventCallback((event: React.FocusEvent) => {
+    const isTargetFromSubtree = Boolean(
+      subtreeRef.current && elementContains(subtreeRef.current, event.target as Node),
+    );
+    if (!isTargetFromSubtree) {
       setActionsVisible(true);
     }
   });
-  const hideActions = useEventCallback((event: React.SyntheticEvent) => {
-    const isEventFromSubtree = subtreeRef.current && elementContains(subtreeRef.current, event.target as Node);
-    if (!isEventFromSubtree) {
-      setActionsVisible(false);
-    }
-  });
 
-  // Listens to focusout event on the document to ensure treeitem actions visibility on portal scenarios
-  // TODO: find a better way to ensure this behavior
-  React.useEffect(() => {
-    if (actionsRef.current) {
-      const handleFocusOut = (event: FocusEvent) => {
-        setActionsVisible(elementContains(actionsRef.current, event.relatedTarget as Node));
-      };
-      targetDocument?.addEventListener('focusout', handleFocusOut, { passive: true });
-      return () => {
-        targetDocument?.removeEventListener('focusout', handleFocusOut);
-      };
+  const handleActionsInvisible = useEventCallback((event: React.FocusEvent) => {
```

Comment:
```suggestion
  const handleActionsInvisible = useEventCallback((event: React.FocusEvent | React.MouseEvent => {
```

You're using these handlers for both mouse and focus events. I'm not more surprised that this doesn't error currently 👀

Reply:
not surprising at all actually, this is because we're using the amazing `getNativeElementProps` that method is basically a blackhole for errors as it doesn't reconcile what it receives with what it returns 😅. but thx for catching the error